src/server/scripts/Outland/zone_netherstorm.cpp

```diff
@@ -699,6 +715,50 @@ class go_captain_tyralius_prison : public GameObjectScript
         }
 };
 
+class npc_captain_tyralius : public CreatureScript
+{
+public:
+    npc_captain_tyralius() : CreatureScript("npc_captain_tyralius") { }
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_captain_tyraliusAI(creature);
+    }
+
+    struct npc_captain_tyraliusAI : public ScriptedAI
+    {
+        npc_captain_tyraliusAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void DoAction(int32 /*action*/) override
+        {
+            _events.ScheduleEvent(EVENT_TALK, 0);
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
```

Comment:
Missing a if (!UpdateVictim()) check here
And this creature should attack something?

Reply:
Nope, it's an always-friendly creature. Never attacks anything.